client.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"log"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+const (
+	pongPeriod = 60 * time.Second
+	pingPeriod = 30 * time.Second
+)
+
+type Client struct {
```

Comment:
I avoided calling it a Channel because the word "channel" has a well-defined semantic meaning in Go, and this definitely isn't it. Perhaps `ClientConnection` is a better name for what this represents (the websocket connection plus the outbound channel... I guess it makes sense).